File: js/Receiver.js
const net = require('net')
const { exit } = require('process')

const HOST = "127.0.0.1"
const PORT = 1234
const URL = "ws://" + HOST + ":" + PORT + "/"
console.log(URL)

let sock = new net.Server()
sock.listen(PORT, HOST)

sock.on("connection", (con) => {
    con.on("data", (data) => {
        obj = JSON.parse(data)
        console.log("received object:")
        console.log(obj)
        con.destroy()
        exit()
    })
})

File: python/Receiver.py
import socket as s
import json
from TestObject import TestObject

HOST = "127.0.0.1"
PORT = 1234

def main():
    # open socket
    sock = s.socket()
    sock.bind((HOST, PORT))
    sock.listen()

    con, addr = sock.accept()
    print(f"binding to: {addr}")
    received_data = bytes()
    while True:
        data = con.recv(1024)
        if not data:
            break
        received_data += data
    con.close()
    sock.close()
	
    unserialized_obj = json.loads(received_data)
    obj = TestObject(**unserialized_obj)
    print("received object:")
    print(obj)

if __name__ == "__main__":
    main()
